src/System.IO.Pipelines/ReadableBufferReader.cs

```diff
@@ -80,7 +86,7 @@ public int Take()
 
         private void MoveNext()
         {
-            if (_enumerator.MoveNext())
+            if (_enumerator.MoveNext() && !_enumerator.Current.IsEmpty)
```

Comment:
Skipping empty spans on MoveNext is fine. We should actually be calling `MoveNext()` from the ctor